ansible/playbooks/roles/preflight/tasks/main.yml

```diff
@@ -30,6 +30,29 @@
 
 - include_tasks: check-os.yml
 
+- include_vars:
+    file: roles/common/vars/main.yml
+    name: common_vars
+
+- name: Validate network configuration
+  when:
+    - common_vars.provider == "any"
+    - common_vars.specification.cloud is undefined
+  block:
+  - name: Assert default route is defined
+    assert:
+      that: ansible_default_ipv4.address is defined
+      fail_msg: "ansible_default_ipv4.address is undefined. Check default routing configuration, read more in troubleshooting document"
+      quiet: true
+
+  - name: Validate if ansible_default_ipv4.address matches address from inventory
+    when: ansible_default_ipv4.address is defined
+    assert:
+      that:
+        - ansible_default_ipv4.address == hostvars[inventory_hostname].ansible_host
+      fail_msg:  "ansible_default_ipv4.address is {{ ansible_default_ipv4.address }} but inventory uses ip: {{ hostvars[inventory_hostname].ansible_host }}. Check default routing configuration, read more in troubleshooting document"
+      quiet: true
```

Comment:
```suggestion
    - name: Assert default route is defined
      assert:
        that: ansible_default_ipv4.address is defined
        fail_msg: >-
          ansible_default_ipv4.address is undefined. Check default routing configuration,
          read more in troubleshooting document
        quiet: true

    - name: Assert ansible_default_ipv4.address matches address from inventory
      assert:
        that: ansible_default_ipv4.address == hostvars[inventory_hostname].ansible_host
        fail_msg: >-
          ansible_default_ipv4.address is {{ ansible_default_ipv4.address }} but inventory uses ip: {{ hostvars[inventory_hostname].ansible_host }}.
          Check default routing configuration, read more in troubleshooting document
        quiet: true
```

Reply:
After investigation i've changed approach to check if there is only one default routing configuration and compare ```ansible_default_ipv4``` value with value from ```inventory``` file for on-prem installations.